### PR TITLE
Feat(91): Write actions

### LINKED ISSUE #91: Implement `Write` actions in contracts
## Overview

Currently, some of the write actions are not available. 

This ticket is to create the necessary hooks for the available write actions. Simulation functions will also be needed.

## Acceptance criteria 

- Approve Mento transfer 
  - [ ] Simulation function
  - [ ] Execute function
- Lock for veMento 
  - [ ] Simulation function
  - [ ] Execute function
- Create proposal 
  - [ ] Simulation function
  - [ ] Execute function
- Vote 
  - [ ] Simulation function
  - [ ] Execute function

### PR DESCRIPTION
**Pull request by RyRy79261**
## Overview

This PR reorganizes the various write actions found within the repo. 

## Testing

Most of these actions still need to be added to their respective flows, as such this is mostly checking for stability and reading for code quality

## Related issues

- closes #91

### CODE REVIEW

File: src/components/_shared/mento-lock/mento-lock.component.tsx [hunk 1]
```diff
@@ -13,18 +13,31 @@ import {
...
+    Locking: { address: lockingAddress },
+  } = useContracts();
+
+  const { data: allowance } = useAllowance({
+    spender: lockingAddress,
+    owner: address,
+  });
+
+  const { approveMento, reset: resetApprove } = useApprove();
+  const { lockMento, isConfirmed, reset: resetLockHook } = useLockMento();
```
Comment (chapati23) [inline src/components/_shared/mento-lock/mento-lock.component.tsx:40]: nit: inconsistent naming between `resetApprove` and `resetLockHook` - looks like either both should have a `Hook` suffix or none

**Review by chapati23 — APPROVED** [review]
just two nits, rest lgtm!

File: src/app/(routes)/proposals/[id]/page.tsx [hunk 2]
```diff
@@ -18,9 +18,9 @@ import {
 import BlockExplorerLink from "@/components/_shared/block-explorer-link/block-explorer-link.component";
 import { Countdown } from "@/components/countdown/countdown.component";
 import { ProposalCurrentVotes } from "@/components/proposal-current-votes/proposal-current-votes.component";
-import ExecutionCode from "./_components/execution-code.component";
-import Participants from "./_components/participants.component";
-import Vote from "./_components/vote.component";
```
[the remark this answers is not in the document]
Comment (RyRy79261) [reply src/app/(routes)/proposals/[id]/page.tsx:23]: The core thing is it makes the reference approach set up in the `tsconfig.json` be the same through out. 

For example, if you have a component library package and a client library, then declaring this way via a root ts for certain aspects lets you pull in said dependencies as declared in the ts config. 

For the most part, it just appears to be nextjs preference. I like using them for the readabilty of what a thing can be just from the app references. For our scope not much is gained but it's a nice practice if scaling or implementing more complex repo structures

File: src/components/_shared/mento-lock/mento-lock.component.tsx (hunk 1 above)
Comment (RyRy79261) [reply src/components/_shared/mento-lock/mento-lock.component.tsx:40]: Good catch, one sec

File: src/app/(routes)/proposals/[id]/page.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (RyRy79261) [reply src/app/(routes)/proposals/[id]/page.tsx:23]: And the folder structure within the actual endpoints could also be simplified, indexes added for example

**Review by denviljclarke — APPROVED** [review]
LGTM  🫡

**Pull request merged**